UI/MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -0,0 +1,427 @@
+//
+//  RouteViewController.swift
+//  Voyage
+//
+//  Created by Minh Nguyen on 2016-08-23.
+//  Copyright © 2016 Mapbox. All rights reserved.
+//
+
+import UIKit
+import Mapbox
+import MapboxDirections
+import Pulley
+import MapboxNavigation
+import SDWebImage
+
+
+class ArrowFillPolyline: MGLPolylineFeature {}
+class ArrowStrokePolyline: ArrowFillPolyline {}
+
+struct RouteControllerNotification {
+    static let didReceiveNewRoute = Notification.Name("RouteControllerDidReceiveNewRoute")
+}
+
+class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDelegate {
+    @IBOutlet weak var mapView: MGLMapView!
+    @IBOutlet weak var recenterButton: UIButton!
+    
+    var routePageViewController: RoutePageViewController!
+    
+    let routeStepFormatter = RouteStepFormatter()
+    
+    var route: Route { return routeController.routeProgress.route }
+    
+    var directions: Directions!
+    var destination: MGLAnnotation!
+    var pendingCamera: MGLMapCamera?
+    
+    weak var routeController: RouteController!
+    
+    var routeTask: URLSessionDataTask?
+    
+    var currentManeuverArrowPolylines: [ArrowFillPolyline] = []
+    var currentManeuverArrowStrokePolylines: [ArrowFillPolyline] = []
+    let distanceFormatter = DistanceFormatter(approximate: true)
+    let secondsBeforeResetTrackingMode:TimeInterval = 25.0
+    
+    var resetTrackingModeTimer: Timer!
+    
+    let webImageManager = SDWebImageManager.shared()
+    var shieldAPIDataTask: URLSessionDataTask?
+    var shieldImageDownloadToken: SDWebImageDownloadToken?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        automaticallyAdjustsScrollViewInsets = false
+        
+        mapView.delegate = self
+        mapView.tintColor = NavigationUI.shared.tintColor
+        recenterButton.applyDefaultCornerRadiusShadow(cornerRadius: 22)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        mapView.compassView.isHidden = true
+        
+        mapView.addAnnotation(destination)
+        
+        if let camera = pendingCamera {
+            camera.altitude = 1_000
+            camera.pitch = 45
```

Comment:
If a developer sends through a camera with a custom pitch and altitude, we should honor these values.